internal/fxclock/clock_test.go

```diff
@@ -22,44 +22,150 @@ package fxclock
 
 import (
 	"context"
+	"sync"
 	"testing"
 	"time"
 
-	"github.com/benbjohnson/clock"
 	"github.com/stretchr/testify/assert"
 )
 
-var _ Clock = clock.Clock(nil)
-
-// Just a basic sanity check that everything is in order.
 func TestSystemClock(t *testing.T) {
-	t.Parallel()
-
 	clock := System
+	testClock(t, System, clock.Sleep)
+}
+
+func TestMockClock(t *testing.T) {
+	clock := NewMock()
+	testClock(t, clock, clock.Add)
+}
 
-	t.Run("Now and Since", func(t *testing.T) {
-		t.Parallel()
+func testClock(t *testing.T, clock Clock, advance func(d time.Duration)) {
+	now := clock.Now()
+	assert.False(t, now.IsZero())
 
-		before := clock.Now()
-		assert.GreaterOrEqual(t, clock.Since(before), time.Duration(0))
+	t.Run("Since", func(t *testing.T) {
+		advance(1 * time.Millisecond)
+		assert.NotZero(t, clock.Since(now), "time must have advanced")
 	})
 
 	t.Run("Sleep", func(t *testing.T) {
```

Comment:
Won't this run into the race you mentioned in the PR summary? I.e., on the off chance advance runs before sleep blocks, wouldn't the Sleep block forever. For `Mock`, isn't this functionality covered by `TestMock_Sleep`?

Reply:
Yeah, you're right! Great catch. Let me fix this.

Reply:
Fixed!